Description: Remove -Werror check which fails in gcc-14.
  This fixes #1074974 by removing the unneeded check which fails
  when building with gcc-14.
Author: Thomas Adam <thomas@fvwm.org>
Forwarded: https://github.com/fvwmorg/fvwm/pull/106
Debian-Bug: https://bugs.debian.org/1074974
Last-Update: 2024-07-25
diff --git a/configure.ac b/configure.ac
index 97ab9e7a1..a5452e152 100644
--- a/configure.ac
+++ b/configure.ac
@@ -218,17 +218,6 @@ fi
 AC_ISC_POSIX
 AC_MINIX
 
-# catch -Werror and similar options when running configure
-AC_TRY_COMPILE([#include <stdio.h>],
-[int i; static j; int *p; char *c;
-  switch (*p = p = *c) { case 0: printf("%Q", c, p); }
-  *c = &i; c = p;
-  while (1 || (unsigned int)3 >= 0 || ((int)-1) == ((unsigned int)1));
-], , AC_MSG_ERROR("
-configure is not able to compile programs with warnings.  Please
-remove all offending options like -Werror from the CFLAGS and
-CPPFLAGS variables and run configure again."))
-
 # check size of some types
 ac_save_CFLAGS="$CFLAGS"
 CFLAGS="$CFLAGS $X_CFLAGS"
